crates/optimism/txpool/src/maintain.rs

```diff
@@ -75,3 +105,103 @@ where
+            let mut to_remove = Vec::new();
+            let mut to_revalidate = Vec::new();
+            for tx in &pool.pooled_transactions() {
+                // Only interop txs have this field set
+                if let Some(interop) = tx.transaction.interop() {
+                    if !interop.is_valid(timestamp) {
+                        // That means tx didn't revalidated during [`OFFSET_TIME`] time
+                        // We could assume that it won't be validated at all and remove it
+                        to_remove.push(*tx.hash());
+                    } else if interop.is_stale(timestamp, OFFSET_TIME) {
+                        // If tx has less then [`OFFSET_TIME`] of valid time we revalidate it
+                        to_revalidate.push(tx.clone())
+                    }
+                }
+            }
+            if !to_revalidate.is_empty() {
+                let checks_stream =
+                    futures_util::stream::iter(to_revalidate.into_iter().map(|tx| {
+                        let supervisor_client = supervisor_client.clone();
+                        async move {
+                            let check = is_valid_cross_tx(
+                                tx.transaction.access_list(),
+                                tx.transaction.hash(),
+                                timestamp,
+                                Some(TRANSACTION_VALIDITY_WINDOW),
+                                // We could assume that interop is enabled, because
+                                // tx.transaction.interop() would be set only in
+                                // this case
+                                true,
+                                Some(supervisor_client.as_ref()),
+                            )
+                            .await;
+                            (tx.clone(), check)
+                        }
+                    }))
+                    .buffered(MAX_SUPERVISOR_QUERIES);
+                futures_util::pin_mut!(checks_stream);
+                while let Some((tx, check)) = checks_stream.next().await {
+                    if let Some(Err(err)) = check {
+                        // We remove only bad transaction. If error caused by supervisor instability
+                        // or other fixable issues transaction would be validated on next state
+                        // change, so we ignore it
+                        if err.is_bad_transaction() {
+                            to_remove.push(*tx.transaction.hash());
+                        }
+                    } else {
+                        tx.transaction.set_interop(TransactionInterop {
+                            timeout: timestamp + TRANSACTION_VALIDITY_WINDOW,
+                        })
+                    }
+                }
+            }
+            if !to_remove.is_empty() {
+                metrics.inc_removed_tx_interop(to_remove.len());
+                let _ = pool.remove_transactions(to_remove);
```

Comment:
nit, we should only track the actual removed interop tx based on the output here

Reply:
done